lib/route-recognizer.js

```diff
@@ -89,13 +89,13 @@ EpsilonSegment.prototype = {
   generate: function() { return ""; }
 };
 
-function parse(route, names, specificity) {
+function parse(allSegments, route, names, specificity) {
   // normalize route as not starting with a "/". Recognition will
   // also normalize.
   if (route.charAt(0) === "/") { route = route.substr(1); }
 
   var segments = route.split("/");
-  var results = new Array(segments.length);
+  //var results = new Array(segments.length);
```

Comment:
I also reread my first comment, there were 3 things here, I commented only on 2 but one of them was wrong comment.  My thoughts were going faster than I was typing.

The lazy pattern to regex and allSegments were the two big gains (lots of variability so can't give exact numbers bot both seemed almost 50%).  The stability of treating specificity as a string, instead of converting to and from a number and wrapping in an object was small but seemed noticeable.  I suspect with a better more realistic sample data it would be different.